### ISSUE TITLE
Megaraid minor issues

### ISSUE THREAD

**Issue by ulmitov** [truenas/py-SMART#56]
Hi,

Noticed some minor issues when working with megaraid.

Assuming this structure:

root@ubuntu20-cuda:~# smartctl --scan-open
/dev/sda -d scsi # /dev/sda, SCSI device
/dev/sdb -d scsi # /dev/sdb, SCSI device
/dev/sdc -d sat # /dev/sdc [SAT], ATA device
/dev/sdd -d scsi # /dev/sdd, SCSI device
/dev/sde -d scsi # /dev/sde, SCSI device
/dev/sdf -d scsi # /dev/sdf, SCSI device
/dev/sdg -d sat # /dev/sdg [SAT], ATA device
/dev/sdh -d scsi # /dev/sdh, SCSI device
/dev/sdi -d scsi # /dev/sdi, SCSI device
/dev/sdj -d sat # /dev/sdj [SAT], ATA device
/dev/sdk -d scsi # /dev/sdk, SCSI device
/dev/sdl -d sat # /dev/sdl [SAT], ATA device
/dev/sdm -d scsi # /dev/sdm, SCSI device
/dev/sdn -d scsi # /dev/sdn, SCSI device
/dev/sdo -d scsi # /dev/sdo, SCSI device
/dev/sdp -d scsi # /dev/sdp, SCSI device
/dev/bus/0 -d megaraid,38 # /dev/bus/0 [megaraid_disk_38], SCSI device
/dev/bus/0 -d megaraid,39 # /dev/bus/0 [megaraid_disk_39], SCSI device
/dev/bus/0 -d megaraid,42 # /dev/bus/0 [megaraid_disk_42], SCSI device
/dev/bus/0 -d megaraid,43 # /dev/bus/0 [megaraid_disk_43], SCSI device
/dev/bus/0 -d megaraid,44 # /dev/bus/0 [megaraid_disk_44], SCSI device
/dev/bus/0 -d megaraid,45 # /dev/bus/0 [megaraid_disk_45], SCSI device
/dev/bus/0 -d megaraid,47 # /dev/bus/0 [megaraid_disk_47], SCSI device
/dev/bus/0 -d megaraid,48 # /dev/bus/0 [megaraid_disk_48], SCSI device
/dev/bus/0 -d megaraid,53 # /dev/bus/0 [megaraid_disk_53], SCSI device
/dev/bus/0 -d megaraid,55 # /dev/bus/0 [megaraid_disk_55], SCSI device
/dev/bus/0 -d megaraid,56 # /dev/bus/0 [megaraid_disk_56], SCSI device
/dev/bus/0 -d megaraid,57 # /dev/bus/0 [megaraid_disk_57], SCSI device
/dev/bus/0 -d sat+megaraid,58 # /dev/bus/0 [megaraid_disk_58] [SAT], ATA device
/dev/bus/0 -d sat+megaraid,59 # /dev/bus/0 [megaraid_disk_59] [SAT], ATA device
/dev/bus/0 -d sat+megaraid,60 # /dev/bus/0 [megaraid_disk_60] [SAT], ATA device
/dev/bus/0 -d sat+megaraid,61 # /dev/bus/0 [megaraid_disk_61] [SAT], ATA device

So the megaraids are all under same self.dev_reference name which is /dev/bus/0
In that case smartctl obeys to provide the --device option "-d" where we should specify which megaraid it should be.

When using pySMART, some places of code do not use the "-d" option so with megaraids it does not work well
For example in:
device.Device._classify
device.smart_health_assement
Smartctl.health

Please add the -d option where it's missing

**Comment by ralequi** [comment]
Ok, i'm cheking...

- `device.Device._classify` -> it is not used anywhere. In fact it won't work as I see on megaraid disks
- `Smartctl.health` + `device.smart_health_assement` -> done. Nice catch

**Comment by ralequi** [comment]
Yeah, for the issue with the megaraid... could you provide the output of `--all` call for any of the megaraid disks with "sat+" ?
I don't ever seen one, and I would like to add to testbase (just in case they have something special... probably don't, but just in case)

**Comment by ralequi** [comment]
But I suppose the parsing might not be correct as they are "ATA" instead of "SCSI".... or am I wrong with that assumption?

**Comment by ralequi** [comment]
> Smartctl.health + device.smart_health_assement -> done. Nice catch -> can this one be moved under device class as a method ?

I don't have to.
You already have `Device.assessment` that it is exactly what `--health` returns

```python
        self.assessment: Optional[str] = None
        """
        **(str):** SMART health self-assessment as reported by the device.
        """
        ``` 
        
 If you wish, I could add a `health()` that returns true on "PASS" and false on "FAIL"

**Comment by ralequi** [comment]
> device.Device._classify -> it is not used anywhere. In fact it won't work as I see on megaraid disks -> Yes, but actually this method is pretty important, it finds the device protocol pretty well. There are two issues with it - it overrides self.interface while actually finds the transport protocol, second issue it is not taking into account when self.interface = megaraid,xx

I think we should think in other solution.
I mean, interface variable is used internally, so probably to handle smartctl calls... Probably having other variable to specify "better" que disk interface should be better...

Now, convert current interface to _interface and use "interface" as the truly disk-interface or create a new variable... don't know which one should be better.

**Comment by ralequi** [comment]
Sorry @ulmitov , busy week, I couln't handle this before.

> Before doing these changes is it possible please to increase the release to 1.2.2 ?
I would like to reduce the number of releases as much as possible. I don't see the point of having too much many versións out there.
In the other hand, these changes should be easy to be included ( let me re-read everything and start with it)

**Comment by ralequi** [comment]
Just to confirm @ulmitov ,

the output of `smartctl --all -d megaraid,61 /dev/bus/0` is exactly the same as the output of `smartctl --all -d sat+megaraid,61 /dev/bus/0` ?

**Comment by ralequi** [comment]
try checking it again

**Comment by ralequi** [comment]
In any case, i'm trying to run the `-l sasphy` on my raids, but never returns what It should.... Don't know about sata on raidcards, but probably won't work either.

The "not really good" check implemented on the interface property should handle this but... don't know how well

**Comment by ralequi** [comment]
> Hey Rafael, Actually it became a bit confusing, when there are two interface attributes, each time being confusing which one is which, can they have different names ? The _interface attr is also a breaking change, maybe keep the previous name?

Maybe you are right, let's change the name...

**Comment by ralequi** [comment]
changes:

interface -> dev_interface
interface -> wraps ( smartctl_interface )

**Comment by ralequi** [comment]
Preparing for v1.2.2 release

**Comment by ralequi** [comment]
Released v1.2.2
